AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1030,10 +1041,10 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
     }
     
     ASPerformBlockOnMainThread(^{
-      // Grab __instanceLock__ here to make sure this transition isn't invalidated
-      // right after it passed the validation test and before it proceeds
-      ASDN::MutexLocker l(__instanceLock__);
-
+      // Hold onto __instanceLock__ until at least after the animation of this transtion kicked off,
+      // to make sure this transition isn't invalidated after it passed the abort test below
+      __instanceLock__.lock();
+      
       if ([self _shouldAbortTransitionWithID:transitionID]) {
         return;
```

Comment:
Good catch! It's very hard to hit this line but it's a legitimate mistake of mine!